test: Post review comments for uncovered code additions

mvollmer opened the pull request:
This also moves all the coverage related code into lcov.py.

This is a continuation of #17475 that just makes comments and doesn't request changes.  Having cockpituous request changes might block merging in annoying ways.  Maybe we want that eventually, but not yet.

cockpituous commented on pkg/shell/hosts.jsx:
```diff
@@ -127,6 +127,7 @@ export class CockpitHosts extends React.Component {
     filterHosts(host, term) {
         if (!term)
             return host;
+        console.log("Executed?");
```
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17523-20220705-064131-58777222-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)

cockpituous commented on pkg/shell/hosts.jsx:
```diff
@@ -119,6 +119,8 @@ export class CockpitHosts extends React.Component {
             this.props.jump(addr);
         }
 
+        console.log("Executed?");
```
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17523-20220705-071122-4dc49800-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)

mvollmer commented:
I did these tests:

- Added a line of uncovered code -> cockpituous pointed it out correctly.
- Moved the line somewhere else -> the old comment was deleted and a new one appeared for the new place
- Removed the line completely -> the second comment disappeared without a trace.
